server/server_test.go

```diff
@@ -27,6 +29,145 @@ func (lb *testLoadBalancer) Servers() []*url.URL {
 	return []*url.URL{}
 }
 
+func TestServerMultipleFrontendRules(t *testing.T) {
+	cases := []struct {
+		expression  string
+		requestURL  string
+		expectedURL string
+	}{
+		{
+			expression:  "Host:foo.bar",
+			requestURL:  "http://foo.bar",
+			expectedURL: "http://foo.bar",
+		},
+		{
+			expression:  "PathPrefix:/management;ReplacePath:/health",
+			requestURL:  "http://foo.bar/management",
+			expectedURL: "http://foo.bar/health",
+		},
+		{
+			expression:  "Host:foo.bar;AddPrefix:/blah",
+			requestURL:  "http://foo.bar/baz",
+			expectedURL: "http://foo.bar/blah/baz",
+		},
+		{
+			expression:  "PathPrefixStripRegex:/one/{two}/{three:[0-9]+}",
+			requestURL:  "http://foo.bar/one/some/12345/four",
+			expectedURL: "http://foo.bar/four",
+		},
+		{
+			expression:  "PathPrefixStripRegex:/one/{two}/{three:[0-9]+};AddPrefix:/zero",
+			requestURL:  "http://foo.bar/one/some/12345/four",
+			expectedURL: "http://foo.bar/zero/four",
+		},
+		{
+			expression:  "AddPrefix:/blah;ReplacePath:/baz",
+			requestURL:  "http://foo.bar/hello",
+			expectedURL: "http://foo.bar/baz",
+		},
+		{
+			expression:  "PathPrefixStrip:/management;ReplacePath:/health",
+			requestURL:  "http://foo.bar/management",
+			expectedURL: "http://foo.bar/health",
+		},
+	}
+
+	for _, test := range cases {
+		router := mux.NewRouter()
+		route := router.NewRoute()
+		serverRoute := &serverRoute{route: route}
+		rules := &Rules{route: serverRoute}
+
+		expression := test.expression //"Host:foo.bar"
+		routeResult, err := rules.Parse(expression)
+
+		if err != nil {
+			t.Fatal("Error while building route for Host:foo.bar")
+		}
+
+		request, err := http.NewRequest("GET", test.requestURL, nil)
+		routeMatch := routeResult.Match(request, &mux.RouteMatch{Route: routeResult})
+
+		if routeMatch == false {
+			t.Log(err)
+			t.Fatal("Rule Host:foo.bar don't match")
+		}
+
+		server := new(Server)
+
+		server.wireFrontendBackend(serverRoute, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if test.expectedURL != r.URL.String() {
+				t.Fatalf("Expected URL for [%s]: %s, got %s", test.expression, test.expectedURL, r.URL.String())
+			}
+			t.Logf("Received request to: %s", r.URL.String())
+		}))
+		serverRoute.route.GetHandler().ServeHTTP(nil, request)
+	}
+}
+
+func TestServerLoadConfigMultipleFrontendRules(t *testing.T) {
```

Comment:
Instead of duplicating lots of code from `TestServerLoadConfigHealthCheckOptions`, can we try to handle both test functions in a single test with separate test cases?

Let me know if you need help.